File: android-sdk-ui/src/main/java/com/braze/ui/contentcards/view/CaptionedImageContentCardView.java
package com.braze.ui.contentcards.view;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.appboy.models.cards.CaptionedImageCard;
import com.appboy.ui.R;
import com.braze.support.StringUtils;

public class CaptionedImageContentCardView extends BaseContentCardView<CaptionedImageCard> {
  public CaptionedImageContentCardView(Context context) {
    super(context);
  }

  private class ViewHolder extends ContentCardViewHolder {
    private final TextView mTitle;
    private final TextView mDescription;
    private final ImageView mCardImage;

    ViewHolder(View view) {
      super(view, isUnreadIndicatorEnabled());
      mCardImage = view.findViewById(R.id.com_braze_content_cards_captioned_image_card_image);
      mTitle = view.findViewById(R.id.com_braze_content_cards_captioned_image_title);
      mDescription = view.findViewById(R.id.com_braze_content_cards_captioned_image_description);
    }

    TextView getTitle() {
      return mTitle;
    }

    TextView getDescription() {
      return mDescription;
    }

    ImageView getImageView() {
      return mCardImage;
    }
  }

  @Override
  public ContentCardViewHolder createViewHolder(ViewGroup viewGroup) {
    View view = LayoutInflater.from(viewGroup.getContext())
        .inflate(R.layout.com_braze_captioned_image_content_card, viewGroup, false);
    setViewBackground(view);
    return new ViewHolder(view);
  }

  @Override
  public void bindViewHolder(ContentCardViewHolder viewHolder, final CaptionedImageCard card) {
    super.bindViewHolder(viewHolder, card);
    ViewHolder captionedImageViewHolder = (ViewHolder) viewHolder;

    setOptionalTextView(captionedImageViewHolder.getTitle(), card.getTitle());
    setOptionalTextView(captionedImageViewHolder.getDescription(), card.getDescription());
    captionedImageViewHolder.setActionHintText(StringUtils.isNullOrBlank(card.getDomain()) ? card.getUrl() : card.getDomain());

    setOptionalCardImage(captionedImageViewHolder.getImageView(), card.getAspectRatio(), card.getImageUrl(), card);
    viewHolder.itemView.setContentDescription(card.getTitle() + " . " + card.getDescription());
  }
}

File: android-sdk-ui/src/main/java/com/braze/ui/contentcards/view/CaptionedImageContentCardView.kt
package com.braze.ui.contentcards.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.appboy.models.cards.CaptionedImageCard
import com.appboy.models.cards.Card
import com.appboy.ui.R

open class CaptionedImageContentCardView(context: Context) : BaseContentCardView<CaptionedImageCard>(
    context
) {
    private inner class ViewHolder constructor(view: View) :
        ContentCardViewHolder(view, isUnreadIndicatorEnabled) {
        val title: TextView? = view.findViewById(R.id.com_braze_content_cards_captioned_image_title)
        val description: TextView? = view.findViewById(R.id.com_braze_content_cards_captioned_image_description)
        val imageView: ImageView? = view.findViewById(R.id.com_braze_content_cards_captioned_image_card_image)
    }

    override fun createViewHolder(viewGroup: ViewGroup): ContentCardViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.com_braze_captioned_image_content_card, viewGroup, false)
        setViewBackground(view)
        return ViewHolder(view)
    }

    override fun bindViewHolder(viewHolder: ContentCardViewHolder, card: Card) {
        if (card is CaptionedImageCard) {
            super.bindViewHolder(viewHolder, card)
            val captionedImageViewHolder = viewHolder as ViewHolder
            captionedImageViewHolder.title?.let { setOptionalTextView(it, card.title) }
            captionedImageViewHolder.description?.let { setOptionalTextView(it, card.description) }
            (if (card.domain.isNullOrBlank()) card.url else card.domain)?.let {
                captionedImageViewHolder.setActionHintText(
                    it
                )
            }
            setOptionalCardImage(
                captionedImageViewHolder.imageView,
                card.aspectRatio,
                card.imageUrl,
                card
            )
            viewHolder.itemView.contentDescription = "${card.title} .  ${card.description}"
        }
    }
}
